src/main/java/com/direwolf20/buildinggadgets/common/building/placement/OriginWrapper.java

```diff
@@ -0,0 +1,36 @@
+package com.direwolf20.buildinggadgets.common.building.placement;
+
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.util.math.BlockPos;
+
+/**
+ * Wraps an {@link IBlockProvider} such that all access to the provider will be translated by the given amount as the
+ * BlockPos passed into the constructor.
+ */
+public final class OriginWrapper implements IBlockProvider {
+
+    private final IBlockProvider provider;
+    private final BlockPos origin;
+
+    /**
+     * @implNote you cannot create a object and pass itself as a parameter of the constructor, so we are safe from looping
+     */
+    public OriginWrapper(IBlockProvider provider, BlockPos origin) {
+        this.provider = provider;
+        this.origin = origin;
+    }
+
+    @Override
+    public BlockPos getOrigin() {
```

Comment:
As it is possible to combine Translation, this should take the underlying Provider's Translation into Account.

Reply:
I think we should consider the wrapped `IBlockProvider` as a black box. What I will do is add a `getHandle` method to access the underlying wrapped block provider

Reply:
That's not what I meant: A user would expect this Method to return the total translation performed. If now you do 2 or more translations in a row, someone, who tries to find out how the result is going to be translated will only see the OuterMostTranslation, but not the total translation. If you'd add a getHandle Method, that would result in one being able to retrace 2 Translations, but still not the Total one.

I agree that the BlockProvider should be a blackbox. Just do provider.getOrigin().add(origin)...